libraries/GCS_MAVLink/MissionItemProtocol.cpp

```diff
@@ -277,6 +277,13 @@ void MissionItemProtocol::transfer_is_complete(const GCS_MAVLINK &_link, const m
     free_upload_resources();
     receiving = false;
     link = nullptr;
+
+    // broadcast our new mission count to let GCSs download it if they wish
```

Comment:
Fixed.

'though I think we should pass this by some other GCS authors.

Reply:
@WickedShell ?

Reply:
You can't send this out on anything that is MAVLink1 unless the mission_type is actually MISSION. If you do this with rally points or fence stuff and it's sent as MAVLink1 you are telling a GCS that it's mission changed when it didn't really, which will lead to GCS confusion.

It's a bit weird to spam everyone, but there also isn't a formal way to notice that a mission changed so I don't have a strong opinion. In practice though a GCS will still have to poll the mission count because there is no other way for us to tell if the mission actually changed or not, and at the same time just because the count changed doesn't tell us anything. And even if we see a count come in on our link it's not clear we should actually start downloading/changing anything, because it might be another component who asked for it that we are forwarding to/mirroring telemetry from.

I don't have anything definitive here, but this is really quite far from ensuring that everyone gets a real notice the mission changed, so I'm not sure it's worth fixing with the complexity of managing link types.

Reply:
That's a really important point that @WickedShell made about MAVLink1 links.

So I thought about this a bit more and I got a question about it: what GCS would (currently) benefit from this broadcast? I looked at MAVProxy, Mission Planner and QGC and from what I can tell, none of them would use this info to do anything.
Actually, it seems there is an issue with sending it: if the GCS isn't expecting the message, it might log or even display an error to the user.

Given this, I think we should remove it and not broadcast it at all.

Reply:
It's been removed.